mlflow/models/model.py

```diff
@@ -863,6 +881,17 @@ def log(
                     f"`{MLFLOW_RECORD_ENV_VARS_IN_MODEL_LOGGING.name}` to `false`."
                 )
             mlflow_model.env_vars = env_vars
+
+            # Associate prompts to the model Run
+            client = mlflow.MlflowClient()
+            for prompt_uri in prompts or []:
+                try:
+                    client.log_prompt(run_id, prompt_uri)
+                except MlflowException:
+                    _logger.warning(
+                        f"Failed to associate prompt {prompt_uri} with the model run {run_id}."
+                    )
```

Comment:
```suggestion
            # Associate prompts to the model Run
            if prompts:
                client = mlflow.MlflowClient()
                for prompt_uri in prompts:
                    try:
                        client.log_prompt(run_id, prompt_uri)
                    except MlflowException:
                        _logger.warning(
                            f"Failed to associate prompt {prompt_uri} with the model run {run_id}."
                        )
```

to skip client creation